dask/dataframe/tests/test_groupby.py

```diff
@@ -3196,3 +3381,101 @@ def test_groupby_slice_getitem(by, slice_key):
     # column projection after read_parquet etc
     assert hlg_layer(got.dask, "getitem")
     assert_eq(expect, got)
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        "min",
+        "max",
+        "sum",
+        "prod",
+        "first",
+        "last",
+        pytest.param(
+            "idxmax",
+            marks=pytest.mark.skip(reason="https://github.com/dask/dask/issues/9882"),
+        ),
+        pytest.param(
+            "idxmin",
+            marks=pytest.mark.skip(reason="https://github.com/dask/dask/issues/9882"),
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "numeric_only",
+    [None, True, False],
+)
+def test_groupby_numeric_only_supported(func, numeric_only):
+    pdf = pd.DataFrame(
+        {
+            "ints": [4, 4, 5, 5, 5],
+            "ints2": [1, 2, 3, 4, 1],
+            "dates": pd.date_range("2015-01-01", periods=5, freq="1T"),
+            "strings": ["q", "c", "k", "a", "l"],
+        }
+    )
+    ddf = dd.from_pandas(pdf, npartitions=3)
+
+    kwargs = {} if numeric_only is None else {"numeric_only": numeric_only}
+
+    # Some groupby methods will raise deprecation warnings or TypeErrors
+    # depending on the version of pandas being used. Here we check that
+    # dask and panadas have similar behavior
+    ctx = contextlib.nullcontext()
+    expected = None
+    try:
+        with warnings.catch_warnings():
+            # This particular warning happens in pandas<1.5,
+            # but Dask only emits a warning with pandas>=1.5,<2.0.
+            warnings.filterwarnings(
+                "ignore",
+                message="Dropping of nuisance columns",
+                category=FutureWarning,
+            )
+            expected = getattr(pdf.groupby("ints"), func)(**kwargs)
+    except FutureWarning:
+        # FutureWarning may have different messages, depending on the value of
+        # numeric_only. For simplicity, we're not checking messages here.
+        # We only warn in pandas>=1.5,<2.0.
+        ctx = pytest.warns(FutureWarning)
+    except TypeError:
+        # Different messages will be raised for the various groupby methods.
+        # To make things simple, we just check that a TypeError is always raised.
+        ctx = pytest.raises(TypeError)
+    with ctx:
+        actual = getattr(ddf.groupby("ints"), func)(**kwargs)
+        if expected is not None:
+            # expected is None if an error was raised
+            assert_eq(expected, actual)
+
+
+# TODO: this test will fail when we implement support for numeric_only=False for the following agg functions
+@pytest.mark.parametrize(
+    "func",
+    ["cumsum", "cumprod", "mean", "median", "var", "std", "corr", "cov"],
```

Comment:
Probably best to use `NUMERIC_ONLY_NOT_IMPLEMENTED` here. That way we only need to update what methods are unsupported in a single place

Reply:
👍